dbms/src/Common/SensitiveDataMasker.h

```diff
@@ -0,0 +1,173 @@
+        uint64_t getMatchesCount() const { return matches_count; }
+    };
+
+    std::vector<std::unique_ptr<MaskingRule>> all_masking_rules;
+
+public:
+    SensitiveDataMasker() {}
+
+    SensitiveDataMasker(const Poco::Util::AbstractConfiguration & config, const std::string & config_prefix)
+    {
+        Poco::Util::AbstractConfiguration::Keys keys;
+        config.keys(config_prefix, keys);
+        Logger * logger = &Logger::get("SensitiveDataMaskerConfigRead");
+
+        std::set<std::string> used_names;
+
+        for (const auto & rule : keys)
+        {
+            if (startsWith(rule, "rule"))
+            {
+                auto rule_config_prefix = config_prefix + "." + rule;
+
+                auto rule_name = config.getString(rule_config_prefix + ".name", rule_config_prefix);
+
+                if (used_names.count(rule_name) == 0)
+                {
+                    used_names.insert(rule_name);
+                }
+                else
+                {
+                    throw Exception(
+                        "query_masking_rules configuration contains more than one rule named '" + rule_name + "'.",
+                        ErrorCodes::INVALID_CONFIG_PARAMETER);
+                }
+
+                auto regexp = config.getString(rule_config_prefix + ".regexp", "");
+
+                if (regexp == "")
+                {
+                    throw Exception(
+                        "query_masking_rules configuration, rule '" + rule_name + "' has no <regexp> node or <regexp> is empty.",
+                        ErrorCodes::NO_ELEMENTS_IN_CONFIG);
+                }
+
+                auto replace = config.getString(rule_config_prefix + ".replace", "******");
+
+                try
+                {
+                    this->addMaskingRule(rule_name, regexp, replace);
+                }
+                catch (DB::Exception & e)
+                {
+                    e.addMessage("while adding query masking rule '" + rule_name + "'.");
+                    throw;
+                }
+            }
+            else
+            {
+                LOG_WARNING(logger, "Unused param " << config_prefix << '.' << rule);
+            }
+        }
+        auto rules_count = this->rulesCount();
+        if (rules_count > 0)
+        {
+            LOG_INFO(logger, rules_count << " query masking rules loaded.");
+        }
+    }
+
+    void addMaskingRule(const std::string & name, const std::string & regexpString, const std::string & replacementString)
+    {
+        all_masking_rules.push_back(std::make_unique<MaskingRule>(name, regexpString, replacementString));
+    }
+
+    int wipeSensitiveData(std::string & data)
+    {
+        int matches = 0;
+        for (auto & rule : all_masking_rules)
+        {
+            matches += rule->apply(data);
+        }
+        return matches;
+    }
+
+#ifndef NDEBUG
+    // TODO: we can add something like system.query_masking_rules with that data
```

Comment:
It will be an overkill, but look at the proposal: https://gist.github.com/alexey-milovidov/d62d73222d83b9319dc519cbb13aeff6